k-distribution/include/kframework/builtin/domains.k

```diff
@@ -190,9 +190,7 @@ module MAP-JAVA-SYMBOLIC [kast, symbolic]
   rule ((K1 |-> V1) MAP) [ K2 <- V2 ] => (K1 |-> V1) (MAP [ K2 <- V2 ]) requires K1 =/=K K2
 
   rule (MAP:Map [ K1 <- V1 ]) [ K2 <- V2 ] => MAP              [ K1 <- V2 ] requires K1  ==K K2
-
-  // potential infinite loop
-  // rule (MAP:Map [ K1 <- V1 ]) [ K2 <- V2 ] => MAP [ K2 <- V2 ] [ K1 <- V1 ] requires K1 =/=K K2
+//rule (MAP:Map [ K1 <- V1 ]) [ K2 <- V2 ] => MAP [ K2 <- V2 ] [ K1 <- V1 ] requires K1 =/=K K2 // potential infinite loop even if `K2 in_keys(MAP)`
```

Comment:
Note that I also found that this rule could lead to an infinite loop, even if we have the additional condition of `K2 in_keys(MAP)`.  A counter example: `M [ K1 <- _ ] [ K2 <- _ ]` where `M` is a variable, and we have both `K1 in_keys(M)` and `K2 in_keys(M)`.

(It is possible to make it finite by having a custom `in_keys` rules, but its performance overhead could be problematic for a large map. So, such rules are better to be put in project-specific lemmas.)